src/naive.py

```diff
@@ -1,16 +1,46 @@
 """Implementation of the naive exact matching algorithm."""
 
 import argparse
+import fasta
+import fastq
 
+def naive(read, genome):
+    length = len(read[1])
+    out = []
+    for i in range(len(genome[1]) - len(read[1]) + 1):
+        current_match_length = 0
+        for j in range(len(read[1])):
+            if genome[1][i + j] == read[1][j]:
+                current_match_length += 1
+            else:
+                current_match_length = 0
+                break
```

Comment:
This is unnecessary. You only check if `current_match_len` is the same as the read length, and it can never be that inside the loop. If you break from the loop, consequently, `current_match_length != length`. It is only when you run through the entire loop and `j` reaches `len(read[1])` that you get `current_match_length == length`. Setting any `current_match_length < length` to zero doesn't affect the code logic one bit.